File: app/src/main/java/com/siva/virtual/AnswerSecurityQuestionActivity.java
package com.siva.virtual;
import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.text.Editable;
import android.text.TextWatcher;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;
import com.siva.virtual.databinding.ActivityAnswerSecurityQuestionBinding;
import com.siva.virtual.util.Utils;
import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

public class AnswerSecurityQuestionActivity extends AppCompatActivity {

    private final List<String> questions = new ArrayList<>();
    private final List<String> answers = new ArrayList<>();
    private ActivityAnswerSecurityQuestionBinding binding;
    private int current_index = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = DataBindingUtil.setContentView(this,R.layout.activity_answer_security_question);

        SharedPreferences sp = getSharedPreferences(Utils.FIRST_LOGIN_DETAILS,MODE_PRIVATE);
        answers.add(sp.getString(Utils.FIRST_ANSWER,null));
        answers.add(sp.getString(Utils.SECOND_ANSWER,null));
        answers.add(sp.getString(Utils.THIRD_ANSWER,null));

        questions.add(getResources().getString(R.string.first_question));
        questions.add(getResources().getString(R.string.second_question));
        questions.add(getResources().getString(R.string.third_question));

        binding.tvSecurityQuestion.setText(questions.get(current_index));

        binding.btnConfirm.setOnClickListener(view -> {
            String answer = Objects.requireNonNull(binding.etAnswer.getText()).toString().trim();
                if(answer.equals(answers.get(current_index)))
                {
                    Intent i = new Intent(AnswerSecurityQuestionActivity.this,PinSetUpActivity.class);
                    i.putExtra("from_where","from_answer");
                    startActivity(i);

                }else{
                    Utils.showErrorLay(AnswerSecurityQuestionActivity.this,binding.answerErrorLay.cardTop,
                            binding.answerErrorLay.errorLay,binding.answerErrorLay.tvInfo,"Entered answer was incorrect!");
                }
        });

        binding.answerErrorLay.fadeLay.setOnClickListener(view -> Utils.hideErrorLay(AnswerSecurityQuestionActivity.this,binding.answerErrorLay.cardTop,
                binding.answerErrorLay.errorLay));

        binding.tvTryAnother.setOnClickListener(view -> {
            binding.etAnswer.setText(null);
            current_index++;
           if(current_index == 3)
               current_index = 0;

            if(current_index <= 2){
                Animation animation = AnimationUtils.loadAnimation(AnswerSecurityQuestionActivity.this,R.anim.slide_in_right);
                binding.tvSecurityQuestion.startAnimation(animation);
                binding.tvSecurityQuestion.setText(questions.get(current_index));
            }

        });

        binding.etAnswer.setText(null);

        binding.etAnswer.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {
            }

            @Override
            public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {
                binding.btnConfirm.setEnabled(charSequence.length() > 0);
            }

            @Override
            public void afterTextChanged(Editable editable) {
            }
        });
    }
}

File: app/src/main/java/com/siva/virtualWallet/AnswerSecurityQuestionActivity.kt
package com.siva.virtualWallet

import android.content.Intent
import android.os.Bundle
import android.text.Editable
import android.text.TextWatcher
import android.view.View
import android.view.animation.AnimationUtils
import androidx.appcompat.app.AppCompatActivity
import com.siva.virtualWallet.databinding.ActivityAnswerSecurityQuestionBinding
import com.siva.virtualWallet.util.Utils
import java.util.*

class AnswerSecurityQuestionActivity : AppCompatActivity() {
    private val questions: MutableList<String> =
        ArrayList()
    private val answers: MutableList<String?> =
        ArrayList()
    private lateinit var binding: ActivityAnswerSecurityQuestionBinding
    private var current_index =
        0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityAnswerSecurityQuestionBinding.inflate(layoutInflater)
        setContentView(binding.root)
        val sp =
            getSharedPreferences(Utils.FIRST_LOGIN_DETAILS, MODE_PRIVATE)
        answers.add(sp.getString(Utils.FIRST_ANSWER, null))
        answers.add(sp.getString(Utils.SECOND_ANSWER, null))
        answers.add(sp.getString(Utils.THIRD_ANSWER, null))
        questions.add(resources.getString(R.string.first_question))
        questions.add(resources.getString(R.string.second_question))
        questions.add(resources.getString(R.string.third_question))

        binding.tvSecurityQuestion.text =
            questions[current_index]
        binding.btnConfirm.setOnClickListener { view: View? ->
            val answer =
                Objects.requireNonNull(binding.etAnswer.text).toString().trim { it <= ' ' }
            if (answer == answers[current_index]) {
                val i =
                    Intent(this@AnswerSecurityQuestionActivity, PinSetUpActivity::class.java)
                i.putExtra("from_where", "from_answer")
                startActivity(i)
            } else {
                Utils.showErrorLay(
                    this@AnswerSecurityQuestionActivity,
                    binding.answerErrorLay.cardTop,
                    binding.answerErrorLay.errorLay,
                    binding.answerErrorLay.tvInfo,
                    "Entered answer was incorrect!"
                )
            }
        }
        binding.answerErrorLay.fadeLay.setOnClickListener { view: View? ->
            Utils.hideErrorLay(
                this@AnswerSecurityQuestionActivity, binding.answerErrorLay.cardTop,
                binding.answerErrorLay.errorLay
            )
        }
        binding.tvTryAnother.setOnClickListener { view: View? ->
            binding.etAnswer.setText(null)
            current_index++
            if (current_index == 3) current_index =
                0
            if (current_index <= 2) {
                val animation =
                    AnimationUtils.loadAnimation(
                        this@AnswerSecurityQuestionActivity,
                        R.anim.slide_in_right
                    )
                binding.tvSecurityQuestion.startAnimation(animation)
                binding.tvSecurityQuestion.text =
                    questions[current_index]
            }
        }
        binding.etAnswer.setText(null)
        binding.etAnswer.addTextChangedListener(object : TextWatcher {
            override fun beforeTextChanged(charSequence: CharSequence, i: Int, i1: Int, i2: Int) {}
            override fun onTextChanged(charSequence: CharSequence, i: Int, i1: Int, i2: Int) {
                binding.btnConfirm.isEnabled =
                    charSequence.length > 0
            }

            override fun afterTextChanged(editable: Editable) {}
        })
    }
}